sklearn/covariance/robust_covariance.py

```diff
@@ -114,11 +115,13 @@ def _c_step(X, n_support, random_state, remaining_iterations=30,
     X_support = X[support]
     location = X_support.mean(0)
     covariance = cov_computation_method(X_support)
+    precision = pinvh(covariance)
```

Comment:
This causes problems if you put in data that has a singular covariance matrix. The following while loop is skipped, precision is never defined, and then it is referenced on line 145, which throws an error. To avoid the extra `pinvh` call, I can wrap the `precision = pinvh(covariance)` in a conditional after the initial determinant is calculated. I'll commit this, please let me know if that's sufficient.